### PR TITLE
Timer

### LINKED ISSUE #22: Implement timer

### PR DESCRIPTION
**Pull request by Mill3nium**

### CODE REVIEW

File: src/client/main.c [hunk 1]
```diff
@@ -22,8 +25,10 @@ int main(int argc, const char *argv[])
 
     Game* game = GetGame();
 
-    FC_Font* font = FC_CreateFont();  
-    FC_LoadFont(font, game->renderer, "res/fonts/ComicSansMS3.ttf", 20, FC_MakeColor(255, 255, 255, 255), TTF_STYLE_BOLD|TTF_STYLE_ITALIC);
+    if (TTF_Init() < 0) { // ej probelm
+        printf("Error\n");
```
Comment (7Hazard) [inline src/client/main.c]: Make better error log

File: src/client/main.c [hunk 2]
```diff
@@ -43,13 +48,10 @@ int main(int argc, const char *argv[])
...
+            
+            RendererTimer(&game->timer);
 
-            { // Draw CORONA ROYALE text
-                static Uint8 r = 0;
-                static float theta = 0.f; theta+=0.03f;
-                r = ((sin(theta)+1)/2)*255;
-                FC_DrawColor(font, game->renderer, 200, 50, FC_MakeColor(r, 20, 20, 255), "CORONA\n%s", "ROYALE");
-            }
+            //lägg in en till sdl renderer
```
Comment (7Hazard) [inline src/client/main.c]: Remove comment

File: src/client/player.c [hunk 3]
```diff
@@ -48,7 +50,8 @@ void HandlePlayerEvents(SDL_Event *event)
             break;
             case SDLK_s: player->down = true;
             break;
-            
+            case SDLK_m: game->timer.mStarted = true;
```
Comment (7Hazard) [inline src/client/player.c]: Remove

File: src/client/timer.c [hunk 4]
```diff
@@ -0,0 +1,117 @@
+#include "timer.h"
+#include <SDL_ttf.h>
+#include <SDL.h>
+#include <SDL_image.h>
+#include "game.h"
+
+void CreateTimer(Timer *timer)
+{
+    TTF_Init(); // va tvungen att lägga den här för att initiera Sans
```
Comment (7Hazard) [inline src/client/timer.c]: Remove, already done in main.c

File: src/client/timer.c [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    if (!timer->Sans)
+    {
+        printf("TTF_OpenFont: %s\n", TTF_GetError());
+        // handle error
+        abort();
+    }
+
+    if (!timer->Sans)
+    {
+        printf("Error %s", TTF_GetError());
```
Comment (7Hazard) [inline src/client/timer.c]: abort or handle error

File: src/client/timer.c [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+        }
+        if (timer->mMinuteTime == 2)
+        {
+            timer->mStarted = false;
+            ResetSeconds;
+        }
+    }
+}
+
+void InitFontTexture(Timer *timer)
```
Comment (7Hazard) [inline src/client/timer.c]: Unused

File: src/client/timer.h [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+    SDL_Rect rect;
+    SDL_Rect position;
+    char timerBuffer[50];
+}Timer;
+
+
+void CreateTimer(Timer* timer);
+void ResetSeconds(Timer* timer);
+void UpdateTimer(Timer* timer);
+void InitFontTexture(Timer* timer);
```
Comment (7Hazard) [inline src/client/timer.h]: Unused

**Review by 7Hazard — CHANGES_REQUESTED** [review]
Fix issues

File: src/client/timer.c (hunk 5 above)
Comment (Mill3nium) [reply src/client/timer.c]: can be more specific about what you mean

Comment (7Hazard) [reply src/client/timer.c]: Either call abort() or handle the function appropriately
Just call abort() in the end if you're unsure about how to handle the error

File: src/client/timer.c (hunk 4 above)
Comment (Mill3nium) [reply src/client/timer.c]: the game crashes without that initialization

Comment (7Hazard) [reply src/client/timer.c]: It should not, this function is already called in main.c

Comment (7Hazard) [reply src/client/timer.c]: Call it inside the GetGame initializer

**Review by 7Hazard — APPROVED** [review]

**Pull request merged**